Resources/Computing/ARCComputingElement.py

```diff
@@ -91,27 +108,38 @@ def submitJob( self, executableFile, proxy, numberOfJobs = 1 ):
     batchIDList = []
     stampDict = {}
 
+    # Do not need to loop over these ... the pilot proxy is the same for everyone
+    usercfg = arc.UserConfig()
+    usercfg.CredentialString(proxy)
+    endpoint = arc.Endpoint(cfgPath("gsiftp://", self.ceHost, ":2811/jobs"), arc.Endpoint.JOBSUBMIT,
+                            "org.nordugrid.gridftpjob")
+
     i = 0
     while i < numberOfJobs:
       i += 1
+      # The basic job description
+      job = arc.Job()
+      jobdescs = arc.JobDescriptionList()
+      # Get the job into the ARC way
       xrslName, diracStamp = self.__writeXRSL( executableFile )
-      cmd = ['arcsub', '-j', self.ceParameters['JobListFile'],
-             '-c', '%s' % self.ceHost, '%s' % xrslName ]
-      result = executeGridCommand( self.proxy, cmd, self.gridEnv )
+      if not arc.JobDescription_Parse(xrslName, jobdescs):
+        logger.msg(arc.ERROR, "Invalid job description")
+        sys.exit(1) 
+      # Submit the job
+      jobs = arc.JobList() # filled by the submit process
+      submitter = arc.Submitter(usercfg)
+      result = submitter.Submit(endpoint, jobdescs, jobs)
+      # Remove clutter
       os.unlink( xrslName )
-      if not result['OK']:
-        break
-      if result['Value'][0] != 0:
-        break
-      pilotJobReference = result['Value'][1].strip()
-      if pilotJobReference and pilotJobReference.startswith('Job submitted with jobid:'):
-        pilotJobReference = pilotJobReference.replace('Job submitted with jobid:','').strip()
+      # Save info or else ...
+      if ( result == arc.SubmissionStatus.NONE ) :
+        # Job successfully submitted
+        pilotJobReference = jobs[0].JobID
         batchIDList.append( pilotJobReference )
         stampDict[pilotJobReference] = diracStamp
-      else:
-        break    
+      else :
```

Comment:
Thanks! Quite useful - will correct in next PR, once I can perform some tests to check that the code is working.